pkg/operator/loglevel/controller.go

```diff
@@ -0,0 +1,128 @@
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	logLevelUtils "github.com/openshift/library-go/pkg/operator/loglevel"
+)
+
+const (
+	controllerName = "loglevel"
+)
+
+// Add creates a new ConfigController and adds it to the Manager.
+func Add(mgr manager.Manager, kubeConfig string) error {
+	r := newReconciler(mgr)
+
+	// Create a new controller
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to CloudCredential object and reconcile the loglevel changes
+	err = c.Watch(
+		&source.Kind{Type: &operatorv1.CloudCredential{}},
+		&handler.EnqueueRequestForObject{})
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	r := &ReconcileCloudCredentialConfig{
+		Client: mgr.GetClient(),
+	}
+	status.AddHandler(controllerName, r)
+
+	return r
+}
+
+var _ reconcile.Reconciler = &ReconcileCloudCredentialConfig{}
+
+// ReconcileCloudCredentialConfig reconciles a CredentialConfig object
+type ReconcileCloudCredentialConfig struct {
+	client.Client
+}
+
+// Reconcile reads the state of the cluster for a CredentialConfig object and
+// makes changes based on the state read and what is in the CredentialConfig.LogLevel
+// Automatically generate RBAC rules to allow the Controller to read and write required types.
+// +kubebuilder:rbac:groups=operator.openshift.io,resources=cloudcredential/spec,verbs=get
+func (r *ReconcileCloudCredentialConfig) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	logger := log.WithFields(log.Fields{
+		"controller": controllerName,
+		"cr":         fmt.Sprintf("%s/%s", request.NamespacedName.Namespace, request.NamespacedName.Name),
+	})
+
+	// Check OperatorLogLevel and LogLevel
+	log.Debug("Checking log level for Reconcile request")
+	desiredOperatorLogLevel, err := utils.GetOperatorLogLevel(r, logger)
+	if err != nil || !logLevelUtils.ValidLogLevel(desiredOperatorLogLevel) {
+		return reconcile.Result{Requeue: false}, fmt.Errorf("the operatorLogLevel defined for the cloudcredential object is invalid")
+	}
+
+	desiredLogLevel, err := utils.GetLogLevel(r, logger)
+	if err != nil || !logLevelUtils.ValidLogLevel(desiredLogLevel) {
+		return reconcile.Result{Requeue: false}, fmt.Errorf("the logLevel defined for the cloudcredential object is invalid")
+	}
+
+	// Pick the logLevel with the highest verbosity level.
+	maxLogLevel := desiredOperatorLogLevel
+	if logLevelUtils.LogLevelToVerbosity(desiredLogLevel) > logLevelUtils.LogLevelToVerbosity(desiredOperatorLogLevel) {
+		maxLogLevel = desiredLogLevel
+	}
+
+	currentLogLevel, errBool := logLevelUtils.GetLogLevel()
+	if errBool {
+		log.Debug("Unable to get the current loglevel. Defaulting the current level to 'Normal'")
+	}
+
+	if currentLogLevel == maxLogLevel {
+		return reconcile.Result{Requeue: false}, nil
```

Comment:
nit: If I am not wrong requeue is false by default so `return reconcile.Result{}, nil` should work the same

Reply:
I have changed this for all of the `reconcile.Result{Requeue: false}, nil`  returns